raft/fsm.hh

```diff
@@ -355,6 +355,15 @@ public:
     bool is_candidate() const {
```

Comment:
Add a paragraph to the commit message (and do the same in the PR cover letter) explaining that we don't want to backport this PR because it only improves error messages.

In general, every PR to master should have a paragraph explaining why it should be backported to particular branches or why it shouldn't be backported at all. Not everyone follows this rule especially if a PR is not backported, but please follow it anyway.

Example of a PR that is not backported: #19847
Example of a PR that is backported: #18900

Reply:
> Add a paragraph to the commit message (and do the same in the PR cover letter) explaining that we don't want to backport this PR because it only improves error messages.

I disagree with that reasoning. I created the issue because I was frustrated with debugging failures with error messages lacking details. And what will be the most common source of issues to debug in the upcoming few months? Probably 6.0, 6.1, and 2024.2.